doc/cascadia/Cascading-Default-Settings.md

```diff
@@ -219,15 +229,33 @@ profiles. The generator will return some sort of result indicating that it wants
 a save operation to occur. The app will then save the `profiles.json` file,
 including these new profiles.
 
-If a dynamic profile generator has determined that a profile should no longer be
+We'll generate these dynamic profiles immediately after applying the defaults.
+When a generator runs, it'll be able to create unique profile GUIDs for each
+source it wants to generate a profile for. It'll attempt find any profiles that

```

Comment:
It shouldn't attempt to find any profiles that exist -- right now, during the "early phase," there shouldn't be any profiles that exist except the ones in the defaults. The ones from the json file have been read from disk, but _not yet turned into profile objects_.

The generator will create a bunch of basics, which are then _strictly overlaid_ with the user's json json tree. The generator doesn't ever need to query the user's store.

Reply:
I think there's one other step you missed. We need to make a first pass on the user's settings to make sure all the profiles end up in the right order. We'll do nothing in that pass other than instantiate a list of profiles with matching GUIDs. When we generate profiles dynamically, their settings will be applied to the profile objects that already exist, as if only to maintain the ordering.

Reply:
That doesn't really matter, we can put profiles in order once we load the user's settings. We're using a vector _right now_ but that's an implementation detail.